### PR TITLE
Haiderali29/signup screen UI #3

### LINKED ISSUE #3: signup-screen-ui
create ui for signup screen

### PR DESCRIPTION
**Pull request by Haiderali29**
Closes #3

### CODE REVIEW

File: lib/screens/signup/components/signup_method_button.dart [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import 'package:playon/all_utils.dart';
+
+class SignUpMethodButton extends StatelessWidget {
```
Comment (ahmadkhandev) [inline lib/screens/signup/components/signup_method_button.dart:3]: rename it to SocialSignupButton

File: lib/screens/signup/signup_screen.dart [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+import 'package:playon/all_utils.dart';
+import 'package:playon/screens/signup/components/onboarding_text_field.dart';
+import 'package:playon/screens/signup/components/signup_method_button.dart';
+import 'package:playon/utils/constants.dart';
+import 'package:playon/utils/my_extensions.dart';
+import 'package:playon/widgets/buttons.dart';
+import 'package:playon/widgets/common_widgets.dart';
+
+class SignupScreen extends StatefulWidget {
+  static const signupUi = '/SignupScreen';
```
Comment (ahmadkhandev) [inline lib/screens/signup/signup_screen.dart:11]: always use routeName

File: lib/screens/signup/signup_screen.dart [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+  @override
+  _SignupScreenState createState() => _SignupScreenState();
+}
+
+class _SignupScreenState extends State<SignupScreen> {
+  final TextEditingController _controllerUsername = TextEditingController();
```
Comment (ahmadkhandev) [inline lib/screens/signup/signup_screen.dart:18]: use Name instead of Username

File: lib/screens/signup/signup_screen.dart [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+                      fieldNameText: 'Password'),
+                  OnBoardingTextField(
+                      controller: _controllerConfirmPassword,
+                      fieldNameText: 'Confirm Password'),
+                  Padding(
+                    padding: const EdgeInsets.only(top: 30),
+                    child: MyElevatedButton(
+                      'Sign Up',
+                      onTap: () {},
+                      width: double.maxFinite,
```
Comment (ahmadkhandev) [inline lib/screens/signup/signup_screen.dart:68]: double.infinity

**Review by ahmadkhandev — CHANGES_REQUESTED** [review]

File: lib/screens/signup/signup_screen.dart [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+                          child: 'Sign In'
+                              .toText(fontSize: 17.0, color: AppTheme.primary))
+                    ],
+                  ),
+                  const VerticalSpacing(of: 50),
+                  'Let\'s Get Started!!!'.toText(fontSize: 32.0),
+                  'Create an account to continue'.toText(
+                      fontSize: 17.0, color: AppTheme.lightBlueGreyColor),
+                  const VerticalSpacing(of: 50),
+                  OnBoardingTextField(
```
Comment (ahmadkhandev) [inline lib/screens/signup/signup_screen.dart:51]: selectedColor should be primary. Look into that

**Review by ahmadkhandev — DISMISSED** [review]

**Review by nooralibutt — CHANGES_REQUESTED** [review]

File: lib/screens/signup/components/caption_text_field.dart [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+class CaptionTextField extends StatelessWidget {
+  final String caption;
+  final TextEditingController controller;
+  final TextInputType keyboardType;
+  final bool obscure;
+  const CaptionTextField(
+      {required this.controller,
+      required this.caption,
+      required this.keyboardType,
+      required this.obscure});
```
Comment (ahmadkhandev) [inline lib/screens/signup/components/caption_text_field.dart:16]: make it not required & false as default

File: lib/screens/signup/components/social_signup_button.dart [hunk 7]
```diff
@@ -0,0 +1,34 @@
+import 'package:playon/all_utils.dart';
+
+class SocialSignupButton extends StatelessWidget {
+  const SocialSignupButton(
+      {required this.imagePath, required this.onTap, Key? key})
+      : super(key: key);
```
Comment (ahmadkhandev) [inline lib/screens/signup/components/social_signup_button.dart:6]: write constructor below and the data members above

**Review by ahmadkhandev — APPROVED** [review]

File: lib/screens/signup/signup_screen.dart [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+import 'package:playon/widgets/common_widgets.dart';
+
+class SignupScreen extends StatefulWidget {
+  static const routeName = '/SignupScreen';
+
+  @override
+  _SignupScreenState createState() => _SignupScreenState();
+}
+
+class _SignupScreenState extends State<SignupScreen> {
```
Comment (nooralibutt) [inline lib/screens/signup/signup_screen.dart:18]: Refactor whole screen

**Review by nooralibutt — CHANGES_REQUESTED** [review]

File: lib/screens/signup/components/corner_button.dart [hunk 9]
```diff
@@ -0,0 +1,21 @@
+import 'package:playon/all_utils.dart';
+
+class CornerButton extends StatelessWidget {
```
Comment (ahmadkhandev) [inline lib/screens/signup/components/corner_button.dart:3]: rename it AlignedTextButton and take alignment as a parameter

**Review by ahmadkhandev — DISMISSED** [review]

**Comment by ahmadkhandev** [comment]
@Haiderali29 use this callback:

`typedef BuildContextCallback = void Function(BuildContext context);`

**Review by nooralibutt — CHANGES_REQUESTED** [review]

**Review by ahmadkhandev — APPROVED** [review]

File: lib/screens/signup/components/signup_form.dart [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+class SignupForm extends StatelessWidget {
+  const SignupForm({Key? key}) : super(key: key);
+  @override
+  Widget build(BuildContext context) {
+    final TextEditingController _controllerName = TextEditingController();
+    final TextEditingController _controllerEmail = TextEditingController();
+    final TextEditingController _controllerPhone = TextEditingController();
+    final TextEditingController _controllerPassword = TextEditingController();
+    final TextEditingController _controllerConfirmPassword =
+        TextEditingController();
```
Comment (nooralibutt) [inline lib/screens/signup/components/signup_form.dart]: these should be outside build method

**Review by nooralibutt — DISMISSED** [review]

**Review by nooralibutt — APPROVED** [review]

**Pull request merged**